src/Renderer/LayeredMaterial.js

```diff
@@ -423,12 +447,15 @@ LayeredMaterial.prototype.getLayerUV = function setLayerUV(index) {
     return this.uniforms.paramLayers.value[index].y;
 };
 
-LayeredMaterial.prototype.setLayerOpacity = function setLayerOpacity(index, opacity) {
-    if (this.uniforms.paramLayers.value[index])
-        { this.uniforms.paramLayers.value[index].w = opacity; }
+LayeredMaterial.prototype.setLayerOpacity = function setLayerOpacity(layer, opacity) {
+    const index = Number.isInteger(layer) ? layer : this.indexOfColorLayer(layer.id);
+    if (this.uniforms.paramLayers.value[index]) {
+        this.uniforms.paramLayers.value[index].w = opacity;
+    }
 };
 
-LayeredMaterial.prototype.setLayerVisibility = function setLayerVisibility(index, visible) {
+LayeredMaterial.prototype.setLayerVisibility = function setLayerVisibility(layer, visible) {
+    const index = Number.isInteger(layer) ? layer : this.indexOfColorLayer(layer.id);
     this.uniforms.visibility.value[index] = visible;
 };
```

Comment:
could we deprecate the index-based api (with a console.warn) ? is it still used elsewhere ?